### ISSUE TITLE
Documention

### ISSUE THREAD

**Issue by Mendel1** [kuangliu/pytorch-retinanet#6]
@kuangliu Could you please tell me why you  use log_softmax to compute the focal loss instead of the sigmod layer mentioned in the paper?Or  I made a mistake in  understanding?

**Comment by Mendel1** [comment]
@kuangliu  I have a project like yours ,but it doesn't work well.The significant diffrences between mine and yous is that I resized all image to 384x512 because I have only one 12G GPU and I used sigmod layer .Should that made trouble?And how about your result?Thanks.

**Comment by kuangliu** [comment]
Yeah, it's a bug, should add a sigmoid layer to normalize the cls_output. 
Notice sigmoid don't output the class probability, just normalize it to avoid numeric instability.
And the final class probability should be given by softmax.

I'll test is myself. Thanks for reporting.

**Comment by Mendel1** [comment]
@kuangliu Thanks very much.

**Comment by kuangliu** [comment]
I tried replacing softmax with only sigmoid. It seems working better. 
I'll look into it carefully and report back later.

**Comment by Mendel1** [comment]
@kuangliu  Thanks for your report. I'm reading your code and It puzzles me why you don't use pretrained resnet?Maybe that will help you train faster,I think.

**Comment by Mendel1** [comment]
Another question about the size of image.:sweat:If the size of image can't be divided by 32,error will occur when flow from top to bottom.(eg.c4=11x11,thus c5=5x5 and it'll result in 11x11 + 10x10 when upsampling.Could you explain how it works to me?Thanks very much.:joy:

**Comment by Mendel1** [comment]
@kuangliu  Have you tested  your net on small dataset?I trained the net on 100 pictures selected from voc2007,it behaved well when computing loss.But when I tried to test on it, all labels are zero after
```
score,labels=cls_pred.max(1)
```
And I found all score for those labels(0) is 0.9835.
By the way,I used sigmoid layer instead of softmax.
I really appreciate your help.

**Comment by kexul** [comment]
> Notice sigmoid don't output the class probability, just normalize it to avoid numeric instability.And the final class probability should be given by softmax.
@kuangliu Could you please give a more detailed explanation? Or the reference? I'm shocked since I've misunderstand it for such a long time that I thought the output of the sigmoid is the probability.
@Mendel1 Do you have some idea?
I'll much appreciate your reply.Thanks in advance.

**Comment by kuangliu** [comment]
Forget what I say... Sigmoid works fine.

**Comment by liuzhongling** [comment]
@

> @kuangliu Have you tested your net on small dataset?I trained the net on 100 pictures selected from voc2007,it behaved well when computing loss.But when I tried to test on it, all labels are zero after
> 
> ```
> score,labels=cls_pred.max(1)
> ```
> 
> And I found all score for those labels(0) is 0.9835.
> By the way,I used sigmoid layer instead of softmax.
> I really appreciate your help.

I meet the same question， how do you solve this problem？

**Comment by liuzhongling** [comment]
@Mendel1 


> @kuangliu Have you tested your net on small dataset?I trained the net on 100 pictures selected from voc2007,it behaved well when computing loss.But when I tried to test on it, all labels are zero after
> 
> ```
> score,labels=cls_pred.max(1)
> ```
> 
> And I found all score for those labels(0) is 0.9835.
> By the way,I used sigmoid layer instead of softmax.
> I really appreciate your help.

I meet the same question， how do you solve this problem？
